lib/node_modules/@stdlib/math/base/special/atan2f/src/main.c

```diff
@@ -0,0 +1,142 @@
+*
+* ```text
+* Copyright (c) 2009 The Go Authors. All rights reserved.
+*
+* Redistribution and use in source and binary forms, with or without
+* modification, are permitted provided that the following conditions are
+* met:
+*
+*    * Redistributions of source code must retain the above copyright
+* notice, this list of conditions and the following disclaimer.
+*    * Redistributions in binary form must reproduce the above
+* copyright notice, this list of conditions and the following disclaimer
+* in the documentation and/or other materials provided with the
+* distribution.
+*    * Neither the name of Google Inc. nor the names of its
+* contributors may be used to endorse or promote products derived from
+* this software without specific prior written permission.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+* "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+* LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+* A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+* OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+* SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+* LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+* DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+* THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+* ```
+*/
+
+#include "stdlib/math/base/assert/is_infinitef.h"
+#include "stdlib/math/base/special/copysignf.h"
+#include "stdlib/number/float32/base/signbit.h"
+#include "stdlib/math/base/assert/is_nanf.h"
+#include "stdlib/math/base/special/atanf.h"
+#include "stdlib/constants/float32/pinf.h"
+#include "stdlib/constants/float32/pi.h"
+
+// MAIN //
+
+/**
+* Computes the angle in the plane (in radians) between the positive x-axis and the ray from `(0,0)` to the point `(x,y)` as a single-precision floating-point number.
+*
+* @param y    `y` coordinate
+* @param x    `x` coordinate
+* @return      angle (in radians)
+*
+* @example
+* float v = stdlib_base_atan2f( 2.0f, 2.0f ); // => stdlib_base_atanf( 1.0f )
+* // returns ~0.785f
+*
+* @example
+* float v = stdlib_base_atan2f( 6.0f, 2.0f ); // => stdlib_base_atanf( 3.0f )
+* // returns ~1.249f
+*
+* @example
+* float v = stdlib_base_atan2f( -1.0f, -1.0f ); // => stdlib_base_atanf( 1.0f ) - π
+* // returns ~-2.356f
+*
+* @example
+* float v = stdlib_base_atan2f( 3.0f, 0.0f ); // => π/2
+* // returns ~1.571f
+*
+* @example
+* float v = stdlib_base_atan2f( -2.0f, 0.0f ); // => -π/2
+* // returns ~-1.571f
+*
+* @example
+* float v = stdlib_base_atan2f( 0.0f, 0.0f );
+* // returns 0.0f
+*
+* @example
+* float v = stdlib_base_atan2f( 3.0f, NaN );
+* // returns NaN
+*
+* @example
+* float v = stdlib_base_atan2f( NaN, 2.0f );
+* // returns NaN
```

Comment:
We have been including only one example in C source files:

```suggestion
* @example
* float v = stdlib_base_atan2f( 2.0f, 2.0f ); // => stdlib_base_atanf( 1.0f )
* // returns ~0.785f
```

Reply:
On it! Thanks @Planeshifter for the review!